app/components/documents/document-card.hbs

```diff
@@ -23,34 +23,36 @@
                 {{/unless}}
               </span>
             </div>
-            <LinkTo @route="document" @model={{this.piece.id}}
-              class="auk-h4" data-test-document-card-name-value>
-              {{this.piece.name}}{{#if this.piece.file}}.{{this.piece.file.extension}}{{/if}}
+            <LinkTo
+              @route="document"
+              @model={{this.piece.id}}
+              class="auk-h4"
+              data-test-document-card-name-value
+            >
+              {{#let (or this.piece.file.derived.extension this.piece.file.extension) as |extension|}}
```

Comment:
This results in an annoying re-rendering. Because `this.piece.file` loads in before `this.piece.file.derived`, we will always show the source extension before the derived extension, even if derived exists. So it's going to be `document name.docx` then `document name.pdf` for basically all files from now on.

Perhaps a task or resource could help here, just so that the data for `this.piece.file` and `this.piece.file.derived` exist before we start rendering. But I think that would mean breaking up the component some more (the previous versions would need to live in a separate component to make the loading tasks work as expected). Not insurmountable, but a big chunk of work just to show an extension without annoying redrawing.*

*: I could be wrong, I haven't tried yet, maybe it's actually quite easy to work around the redrawing.

Reply:
@sergiofenoll I forgot about it, but I've created a separate component already in [feature/KAS-3505-draft-access-level-pills-bis ](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1526). Triggered Jenkins once more for that branch, but hope to merge it today.

Reply:
https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1526 is merged. Maybe best to merge development in this PR before making changes.